### PR TITLE
Add Redux Toolkit and Implement Ducks Pattern

### PR DESCRIPTION
**Pull request by the-faizmohammad**
**Description**:
This pull request aims to introduce Redux Toolkit to our project and implements the Ducks pattern for organizing Redux logic. It includes the following changes:

1. Added Redux Toolkit to project dependencies (`npm install react-redux @reduxjs/toolkit`).

2. Created a dedicated directory (`/src/redux`) to contain all Redux-related logic.

3. Configured a Redux store in `/src/redux/store.js`.

4. Defined a slice of state for books in `/src/redux/books/booksSlice.js`:
   - Includes an array of books with an initial state of an empty array.
   - Includes reducers:
     - `addBook` that adds a book to the state.
     - `removeBook` that removes a book from the state.

5. Defined a slice of state for categories in `/src/redux/categories/categoriesSlice.js`:
   - Includes an array of categories with an initial state of an empty array.
   - Includes a reducer that checks the status and always returns "Under construction" as the initial state.

**Additional Information**:
- The project structure has been updated to follow the "feature folder" approach, making it easier to maintain and organize our Redux logic.

**Testing**:
- I have verified that the Ducks pattern has been correctly implemented and that actions and reducers are appropriately organized within their respective slice files.

**Checklist**:
- [ ] I have followed the project's coding conventions and style guide.
- [ ] I have reviewed and formatted the code for readability.
- [ ] I have ensured that there are no console errors or warnings.

Kindly review my pull request. 

Regards,
Faiz

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: 'Under construction',
```
Comment (julie-ify) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly fix your project title on the readme file.

  image

- [ ] This `categories` should be an empty array per the guidelines.

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+};
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
```
Comment (julie-ify) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly implement the categories reducer.

  ```
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @the-faizmohammad,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (the-faizmohammad) [reply src/redux/categories/categoriesSlice.js]: I have implemented required changes to be done.

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (the-faizmohammad) [reply src/redux/categories/categoriesSlice.js]: Aknowledged.
Thank you so much for your feedback.

**Review by danmainah — APPROVED** [review]
Hi @the-faizmohammad , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to ensure the categoriesSlice is properly configured.
✔️  Implemented the change requested to fix the title error in the readme file.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — COMMENTED** [review]
### Hi @the-faizmohammad 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Highlight
🔆 `@reduxjs/toolkit` & `react-redux` is present in `package.json`
🔆 The structure of the files is `ducks pattern`
🔆 Creates 2 reducers, `addBook` and `removeBook`

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — COMMENTED** [review]
Hi @the-faizmohammad, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

dwayne-the-rock

## (Highlights) Good Points: 💪 
🔍 Overall, the student's project fulfills the specified requirements. Here are the highlights and feedback:

Package.json:
✅ The project includes the required dependencies: "@reduxjs/toolkit" and "react-redux".

File Structure and Ducks Pattern:
✅ The file structure follows the ducks pattern, with the following key files:

/src/redux/store.js
/src/redux/books/booksSlice.js
/src/redux/categories/categoriesSlice.js
/src/redux/store.js:
✅ The file correctly imports configureStore from "@reduxjs/toolkit".
✅ The file imports the "books" and "categories" reducers from their respective paths.
✅ configureStore is used to create the reducer, including the "books" and "categories" reducers.

/src/redux/books/booksSlice.js:
✅ The file has been created at /src/redux/books/booksSlice.js (or a similar location).
✅ createSlice is imported from "@reduxjs/toolkit".
✅ An initialState is defined with a "books" state value containing an empty array.
✅ Two reducers, addBook and removeBook, have been created.
✅ The file correctly exports booksSlice.reducer as the default export.
✅ addBook and removeBook are exported as actions.

/src/redux/categories/categoriesSlice.js:
✅ The file has been created at /src/redux/categories/categoriesSlice.js (or a similar location).
✅ createSlice is imported from "@reduxjs/toolkit".
✅ An initialState is defined with a "categories" state value containing an empty array.
✅ A single reducer, checkStatus, has been created.
✅ The file correctly exports categoriesSlice.reducer as the default export.
✅ checkStatus is exported as an action.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**